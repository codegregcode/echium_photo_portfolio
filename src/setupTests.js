import { render, screen, fireEvent, waitFor } from '@testing-library/react';

import { BrowserRouter, BrowserRouter as Router } from 'react-router-dom';

import { describe, it, expect, vi } from 'vitest';

import '@testing-library/jest-dom';

export {
  render,
  screen,
  fireEvent,
  waitFor,
  describe,
  it,
  expect,
  vi,
  Router,
  BrowserRouter,
};
