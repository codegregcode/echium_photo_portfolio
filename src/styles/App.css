.app-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 15px;
  min-height: 100vh;
}

.content {
  flex: 1;
}

.footer {
  padding: 10px;
  text-align: center;
  cursor: pointer;
}

.footer:hover {
  text-decoration: underline;
  color: rgb(0, 255, 197);
}

.modal {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
}

.modal-content {
  background-color: #fefefe;
  margin: auto;
  padding: 20px;
  border: 1px solid #888;
  width: 80%;
  max-width: 500px;
  text-align: center;
}

.close {
  color: #aaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.close:hover,
.close:focus {
  color: black;
  text-decoration: none;
  cursor: pointer;
}
