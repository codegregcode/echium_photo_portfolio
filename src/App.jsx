import { useState } from 'react';
import Lightbox from 'yet-another-react-lightbox';
import { Fullscreen } from 'yet-another-react-lightbox/plugins';
import 'yet-another-react-lightbox/styles.css';
import 'yet-another-react-lightbox/plugins/thumbnails.css';

function App() {
  const [isOpen, setIsOpen] = useState(false);
  const [slideIndex, setSlideIndex] = useState();

  const slides = [
    {
      src: 'https://i.ibb.co/CzFNcWh/793039490023.jpg',
      caption: 'Image 1',
    },
    {
      src: 'https://i.ibb.co/BKXK14y/793039490025.jpg',
      caption: 'Image 2',
    },
    {
      src: 'https://i.ibb.co/Gxvzd4Y/793039490001.jpg',
      caption: 'Image 3',
    },
    {
      src: 'https://i.ibb.co/ysx2dcj/PRINT-17-17-03-24-Leeds-Uni-Circular-Walk-Pentax35-AFM-HP5-400.jpg',
      caption: 'Image 4',
    },
  ];

  const openLightbox = (index) => {
    setSlideIndex(index);
    setIsOpen(true);
  };

  return (
    <div>
      {slides.map((img, index) => (
        <img
          className="thumbnails"
          src={img.src}
          key={img.caption}
          onClick={() => openLightbox(index)}
        />
      ))}

      <Lightbox
        open={isOpen}
        close={() => setIsOpen(false)}
        index={slideIndex}
        slides={slides}
        plugins={[Fullscreen]}
      />
    </div>
  );
}

export default App;
