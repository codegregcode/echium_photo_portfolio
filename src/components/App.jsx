import { Routes, Route } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';

import Welcome from './Welcome';
import Index from './Index';
import Footer from './Footer';

import Mar172024 from './journals/twenty24/Mar172024';
import Apr272024 from './journals/twenty24/Apr272024';
import May2024 from './journals/twenty24/May2024';
import Jun2024 from './journals/twenty24/Jun2024';

import '../styles/App.css';
import 'react-toastify/dist/ReactToastify.css';

function App() {
  return (
    <div className="app-container">
      <ToastContainer />
      <div className="content">
        <Routes>
          <Route path="/" element={<Welcome />} />
          <Route path="/index" element={<Index />} />
          <Route path="/index/a-test-a-start" element={<Mar172024 />} />
          <Route path="/index/porto" element={<Apr272024 />} />
          <Route path="/index/testing-the-a1" element={<May2024 />} />
          <Route path="/index/starting-to-develop" element={<Jun2024 />} />
        </Routes>
      </div>
      {sessionStorage.getItem('show_landing_animation') === '1' &&
      sessionStorage.getItem('show_journal_animation') === '1' ? (
        <Footer />
      ) : null}
    </div>
  );
}

export default App;
