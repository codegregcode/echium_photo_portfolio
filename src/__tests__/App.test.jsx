import { render, BrowserRouter, describe, it, expect } from '../setupTests';
import App from '../components/App';

describe('App component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
