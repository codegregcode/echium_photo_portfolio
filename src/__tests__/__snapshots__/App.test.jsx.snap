// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-container"
  >
    <div
      class="content"
    >
      <div
        class="welcome-component"
      >
        <a
          href="/index"
        >
          <div
            class="welcome-container"
          >
            <h1
              class="welcome-text"
            >
              echium_photo
            </h1>
            <img
              alt="Welcome Image - Click Me"
              class="welcome-background"
              src="https://res.cloudinary.com/dfacllhgp/image/upload/v1722867682/430299530011_mdrw8l.jpg"
              title="click me"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
